src/NServiceBus.AcceptanceTests/DelayedDelivery/TimeoutManager/When_using_external_timeout_manager.cs

```diff
@@ -0,0 +1,83 @@
+﻿namespace NServiceBus.AcceptanceTests.DelayedDelivery
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using AcceptanceTesting.Customization;
+    using EndpointTemplates;
+    using Features;
+    using NServiceBus.Config;
+    using NUnit.Framework;
+
+    public class When_using_external_timeout_manager : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_delay_delivery()
+        {
+            var delay = TimeSpan.FromMilliseconds(1);
+
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<EndpointWithTimeoutManager>()
+                .WithEndpoint<Endpoint>(b => b.When((session, c) =>
+                {
+                    var options = new SendOptions();
+
+                    options.DelayDeliveryWith(delay);
+                    options.RouteToThisEndpoint();
+
+                    c.SentAt = DateTime.UtcNow;
+
+                    return session.Send(new MyMessage(), options);
+                }))
+                .Done(c => c.WasCalled)
+                .Run();
+
+            Assert.GreaterOrEqual(context.ReceivedAt - context.SentAt, delay);
```

Comment:
is a millisecond really a reliable time to be sure it was dispatched using a timeout manager? How about checking the presence of headers set by the timeout manager?